### PR TITLE
sql/seqexpr: Remove dep from seqexpr to builtin

### PR DESCRIPTION
**Pull request by Xiang-Gu**
`seqexpr` package is a low level package that manipulates
sequence expressions and we do not want it to depend on
the "heavy" package `builtins`.

To break this dependency, we had two commits for this PR:

1. Created a new package `builtinconstants` and moved 
    all constants used in the builtin package there.

2. We plumbed in the only function call from the `builtins`
    package in `seqexpr` to remove the dependency from
    seqexpr to builtins. Finally, we added a bazel rule in
    seqexpr/BUILD.bazel to disallow dependency to builtins

Release note: None

### CODE REVIEW

**Review by chengxiong-ruan — APPROVED** [review]
lgtm

File: pkg/sql/catalog/seqexpr/sequence.go [hunk 1]
```diff
@@ -42,7 +42,10 @@ func (si *SeqIdentifier) IsByID() bool {
 // takes a sequence identifier as an arg (a sequence identifier can either be
 // a sequence name or an ID), wrapped in the SeqIdentifier type.
 // Returns the identifier of the sequence or nil if no sequence was found.
-func GetSequenceFromFunc(funcExpr *tree.FuncExpr) (*SeqIdentifier, error) {
+func GetSequenceFromFunc(
+	funcExpr *tree.FuncExpr,
+	getBuiltinProperties func(name string) (*tree.FunctionProperties, []tree.Overload),
+) (*SeqIdentifier, error) {
```
Comment (ajwerner) [inline pkg/sql/catalog/seqexpr/sequence.go:48]: nit: comment around these functions that this argument is commonly `builtins.GetBuiltinProperties`?

**Review by ajwerner — APPROVED** [review]
LGTM

File: pkg/sql/sem/builtins/builtinconstants/constants.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package builtinconstants
+
+import "time"
+
+// SequenceNameArg represents the name of sequence (string) arguments in
+// builtin functions.
+const SequenceNameArg = "sequence_name"
```
[the remark this answers is not in the document]
Comment (chengxiong-ruan) [reply pkg/sql/sem/builtins/builtinconstants/constants.go:17]: I thought it's just used to extract sequence name from expressions, no?

[the remark this answers is not in the document]
Comment (ajwerner) [reply pkg/sql/sem/builtins/builtinconstants/constants.go:17]: Yeah, that's true, but only for the case of this weird overload, right?

**Comment by Xiang-Gu** [comment]
TFTR
bors r+

**Pull request merged**